File: backend/src/models/Photo.js
const mongoose = require('mongoose');
const Schema = mongoose.Schema;

const PhotoSchema = new Schema({
    url: { type: String, required: true },
    publicId: { type: String },
    group: { type: Schema.Types.ObjectId, ref: 'Group' },
    uploader: { type: Schema.Types.ObjectId, ref: 'Photographer' },
    filename: String,

    // --- CHANGED: Support Multiple Faces ---
    // Was: embedding: { type: [Number], default: [] }
    // Now: Array of Arrays (List of faces)
    embeddings: { type: [[Number]], default: [] }, 
    // ---------------------------------------

    createdAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model('Photo', PhotoSchema);

File: backend-java/src/main/java/com/kwikpic/model/Photo.java
package com.kwikpic.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

/**
 * Maps to Mongoose Photo schema in models/Photo.js.
 *
 * The embeddings field is List&lt;List&lt;Double&gt;&gt; — each inner list is a
 * 512-float vector representing one face found in the photo.
 * Mongoose schema: embeddings: { type: [[Number]], default: [] }
 */
@Document(collection = "photos")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Photo {

    @Id
    private String id;

    private String url;

    /** Cloudinary public ID for management/deletion. */
    private String publicId;

    /** ObjectId hex-string referencing Group. */
    private String group;

    /** ObjectId hex-string referencing Photographer (uploader). */
    private String uploader;

    private String filename;

    /**
     * Core AI field. Each inner List is a 512-float embedding vector for one face.
     * A photo with 3 people yields 3 inner lists.
     */
    private List<List<Double>> embeddings = new ArrayList<>();

    private Date createdAt;
}
